### ISSUE TITLE
[flake8-type-checking] TC010 false positive

### ISSUE THREAD

**Issue by dacevedo12** [astral-sh/ruff#15162]
Given a recursive type hint like the following:

```python
TMetadata = dict[str, int | "TMetadata"]
```

Ruff says `Invalid string member in `X | Y`-style union type Ruff(TC010)`

As a workaround, one can declare the union using the old way

```python
TMetadata = dict[str, Union[int, "TMetadata"]]
```

**Comment by dacevedo12** [comment]
Seems to be a limitation in python then. It looks like 3.12's `type` statement will solve this by removing the need for reference strings
